spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -274,18 +275,21 @@ public boolean canDeleteWhere(Filter[] filters) {
       }
     }
 
-    return deleteExpr == Expressions.alwaysTrue() || canDeleteUsingMetadata(deleteExpr);
+    return selectsPartitions(deleteExpr) || canDeleteUsingMetadata(deleteExpr);
+  }
+
+  private boolean selectsPartitions(Expression expr) {
```

Comment:
It is a similar trick to what we do in `SparkScanBuilder`.

Reply:
+1, if selectsPartitions returns true and it's an disjunction, so canDeleteUsingMetadata will not be called.

[question] if we can move selectsPartition into an util, for ex: SparkUtil we can make SparkScanBuilder use the same. As in one place i.e here we are checking if all are true, and other i.e SparkScanBuilder if atleast one is false.

Reply:
Added a utility to check if an expression selects partitions in all specs.